pandas/core/indexes/datetimes.py

```diff
@@ -684,6 +678,11 @@ def _values(self):
         else:
             return self.values
 
+    @cache_readonly
+    def tz(self):
+        # GH 18595
```

Comment:
i believe we now have this in setter / getter versions

Reply:
Hmm, the only caching decorator I could find was implemented here:
https://github.com/pandas-dev/pandas/blob/c7af4ae94092110b1872750d89ce55bff131b39f/pandas/_libs/properties.pyx#L9-L44